pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -94,39 +94,42 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	// Fetch the BuildRun instance
 	buildRun := &buildv1alpha1.BuildRun{}
 	err := r.client.Get(context.TODO(), request.NamespacedName, buildRun)
-	if err != nil && !errors.IsNotFound(err) {
-		reqLogger.Error(err, "Failed to get the build instance")
+	if err != nil && !apierrors.IsNotFound(err) {
 		return reconcile.Result{}, err
-	} else if errors.IsNotFound(err) {
+	} else if apierrors.IsNotFound(err) {
 		return reconcile.Result{}, nil
 	}
 
 	// Fetch the Build instance
 	build := &buildv1alpha1.Build{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: buildRun.Spec.BuildRef.Name, Namespace: buildRun.Namespace}, build)
 	if err != nil {
-		reqLogger.Error(err, "Failed to get Build from BuildRun", "Build", buildRun.Spec.BuildRef.Name)
+		err = r.updateBuildRunErrorStatus(buildRun, err)
 		return reconcile.Result{}, err
 	}
 
 	lastTaskRun, err := r.retrieveTaskRun(build, buildRun)
 	if err != nil {
-		reqLogger.Error(err, "Failed to list existing TaskRuns from BuildRun", "BuildRun", buildRun.Name)
+		err = r.updateBuildRunErrorStatus(buildRun, err)
```

Comment:
nice func(), however `err` in this line, overrides err in line 112, can we make sure we return both errors somehow?